src/pages/sweepstake/index.jsx

```diff
@@ -0,0 +1,298 @@
+        .estimateGas()
+        
+        let res = await contractSweepstake.methods
+        .stake(nftID, ethers.utils.parseEther(amount))
+        .send({ from: address, gasLimit })
+
+        // Test to now if the "res.status" return the status success or error of transacion
+        if(res.status == true){
+          sweetAlertSuccess('Success', 'Stake on NFT ' + nftID + ' with amount ' + amount + ' was successful')
+        }else{
+          sweetAlertError('Error', 'Stake on NFT ' + nftID + ' with amount ' + amount + ' was not successful')
+        }
+        
+        getAllSweepstakesContract()
+      })
+    }catch(e){
+        console.log(e)
+    }
+  }
+
+  async function beforeStake(nftID, token){
+    const { value: amount } = await Swal.fire({
+        title: token,
+        input: 'text',
+        inputLabel: 'Amount to stake',
+        inputPlaceholder: '0.00',
+
+        preConfirm: async (amount) => {   
+
+            const contractERC20 = new kit.web3.eth.Contract(ERC20, 
+              token == 'pEUR' ? process.env.NEXT_PUBLIC_CEUR :
+              token == 'pUSD' ? process.env.NEXT_PUBLIC_CUSD :
+              token == 'pBRL' ? process.env.NEXT_PUBLIC_CBRL :
+              null
+              )
+            
+            let amountOfToken;
+            let amountOfAllowance;
+            
+            await performActions(async (kit) => {        
+              const res = await contractERC20.methods.balanceOf(address).call();
+              amountOfToken = ethers.utils.formatEther(res)
+            })
+
+            await performActions(async (kit) => {        
+              const res = await contractERC20.methods.allowance(address, process.env.NEXT_PUBLIC_ESOLIDAR_SWEEPSTAKE).call();
+              amountOfAllowance = ethers.utils.formatEther(res)
+            })
+            
+            amount = amount.replace(',','.')    
+              
+            if(amount.match(/[^0-9.]/g)){
+                sweetAlertError("Error", "Please enter a valid amount")
+                return;
+            }
+            if(amount.match(/[a-z]/i)){
+                sweetAlertError("Error", "Please enter a valid amount")
+                return;
+            }
+
+            try{
+                if(Number(amount) > Number(amountOfToken)){
+                    sweetAlertError("Insufficient Balance", "You need to have more balance to stake")
+                    return
+                }
+                if(Number(amount) > Number(amountOfAllowance)){
+                    addMoreAllowance(amount, contractERC20, amountOfAllowance, nftID) 
+                    return
+                }
+
+                StakeNFT(nftID, amount)                    
+            }catch(e){
+                console.table(e)
+            }
+            
+        }
+      })
+  }
+
+  async function addMoreAllowance(amount, token, currentAllowance, nftID){
+    
+    let remaining = amount - currentAllowance
```

Comment:
Codacy has a fix for the issue: ['remaining' is never reassigned. Use 'const' instead.](https://app.codacy.com/gh/esolidar/web3/pullRequest?prid=9542591)
```suggestion
    const remaining = amount - currentAllowance
```